channels/channelmanager.go

```diff
@@ -0,0 +1,108 @@
+package channels
+
+import (
+	"github.com/ExchangeUnion/xud-tests/database"
+	"github.com/ExchangeUnion/xud-tests/discord"
+	"github.com/ExchangeUnion/xud-tests/xudrpc"
+	"github.com/google/logger"
+	"time"
+)
+
+type ChannelManager struct {
+	Interval int `long:"manager.interval" default:"20" description:"Interval in seconds at which new channels should be opened"`
+
+	channels []*Channel
+
+	xud      *xudrpc.Xud
+	discord  *discord.Discord
+	database *database.Database
+}
+
+type Channel struct {
+	Currency   string
+	Amount     int64
+	PushAmount int64
+}
+
+func (manager *ChannelManager) Init(channels []*Channel, xud *xudrpc.Xud, discord *discord.Discord, database *database.Database) {
+	logger.Info("Initializing channel manager")
+
+	manager.channels = channels
+
+	manager.xud = xud
+	manager.discord = discord
+	manager.database = database
+
+	ticker := time.NewTicker(time.Duration(manager.Interval) * time.Second)
+
+	manager.openChannels()
+
+	for range ticker.C {
+		manager.openChannels()
+	}
+}
+
+func (manager *ChannelManager) openChannels() {
+	peers, err := manager.xud.ListPeers()
+
+	if err != nil {
+		message := "Could not get XUD peers: " + err.Error()
+
+		logger.Warning(message)
+		_ = manager.discord.SendMessage(message)
+		return
+	}
+
+	for _, peer := range peers.Peers {
+		channelsOpened := manager.database.GetChannelsOpened(peer.NodePubKey)
```

Comment:
I'm thinking here that we may want to add some logic eventually to check if the channels we opened *still* exist. From the looks of it, if we add a channel to the database, then that channel gets closed (maybe by the remote peer), then we won't open another channel. Not a huge concern since other nodes shouldn't be closing channels in simnet all that often, but still something to think about. 

As an aside this makes me think that it'd be nice to have a direct and easy way to query xud for a specific node (we already have `GetNodeInfo` that we can enhance) to see what direct channels we have with that node. Something like that could be handy here.

Reply:
You are right but we don't expect simnet clients to close channels and if they do for whatever reason, it is really easy to create a new identity and get new channels.

Once we have gRPC methods in XUD that allow for a more sophisticated approach, I can revisit this logic.